aeon/schema/social.py

```diff
@@ -0,0 +1,88 @@
+"""Readers for data relevant to Social experiments."""
+
+import json
+from pathlib import Path
+
+import pandas as pd
+
+from aeon import util
+import aeon.io.reader as _reader
+
+
+class Pose(_reader.Harp):
+    """Reader for Harp-binarized tracking data given a model that outputs id, parts, and likelihoods."""
+
+    def __init__(self, pattern):
+        self.pattern = pattern
+        self.extension = "bin"
+
+    def read(self, file, ceph_proc_dir="/ceph/aeon/aeon/data/processed"):
+        """Reads data from the Harp-binarized tracking file."""
+        # Get config file from `file`, then bodyparts from config file.
+        model_dir = file.stem.replace("_", "/")
+        config_file_dir = Path(ceph_proc_dir + "/" + model_dir)
+        assert config_file_dir.exists(), f"Cannot find model dir {config_file_dir}"
+        config_file = get_config_file(config_file_dir)
+        parts = self.get_bodyparts(config_file)
+        # Using bodyparts, assign column names to Harp register values, and read data in default format.
+        columns = ["class", "class_likelihood"]
+        for part in parts:
+            columns.extend([f"{part}_x", f"{part}_y", f"{part}_likelihood"])
+        self.columns = columns
+        data = super().read(file)
+        # Set new columns, and reformat `data`.
+        new_columns = ["class", "class_likelihood", "part", "part_likelihood", "x", "y"]
+        new_data = pd.DataFrame(columns=new_columns)
+        part_data_list = [None] * len(parts)
+        for i, part in enumerate(parts):
+            part_columns = ["class", "class_likelihood", f"{part}_x", f"{part}_y", f"{part}_likelihood"]
+            part_data_list[i] = data[part_columns]
+            part_data_list[i].insert(2, "part", part)
+            part_data_list[i].columns = new_columns
```

Comment:
Here I would probably just take a temp variable for `data[part_columns]` and then manipulate that object directly before using `append` to add it to the list.

Reply:
again, can see my above comment about performancy of list appends in python.

Reply:
Considering the above I would probably have written the whole concatenate with a comprehension.

Reply:
I've seen this pattern of initialising lists in many places, so does seem to be a common Python pattern.

Can you provide an example of how you'd do this with a comprehension?

I think if you want to use a comprehension, you'd want to include the insertion of the `"part"` column and the assignment of `new_columns` in the comprehension, and for this I think you'd need a lambda, and this wouldn't be more performant and for me I don't think it would be more readable. But maybe you're thinking of another way to write this?

Reply:
Ok, set this temp var